dissect/target/plugins/os/unix/linux/fortios/_os.py

```diff
@@ -94,20 +155,101 @@ def ips(self) -> list[str]:
-            with config_fh:
-                return config_fh.readline().split("=")[1]
-
+        """Return FortiOS version"""
+        if self._version:
+            return parse_version(self._version)
         return "FortiOS Unknown"
 
     @export(property=True)
     def os(self) -> str:
-        return OperatingSystem.FORTIGATE.value
+        """Return `fortios`"""
+        return OperatingSystem.FORTIOS.value
+
+    @export(record=FortiOSUserRecord)
+    def users(self) -> Iterator[FortiOSUserRecord]:
+        """Return local users of the FortiOS system"""
+
+        # Administrative users
+        for username, entry in self._config["global-config"]["system"]["admin"].items():
+            yield FortiOSUserRecord(
+                name=username,
+                password=":".join(entry.get("password", [])),
+                groups=entry["accprofile"][0],
+                home="/root",
+                _target=self.target,
+            )
+
+        # Local users
+        local_groups = local_groups_to_users(self._config["root-config"]["user"]["group"])
+        for username, entry in self._config["root-config"]["user"].get("local", {}).items():
+            try:
+                password = decrypt_password(entry["passwd"][-1])
+            except ValueError:
+                password = ":".join(entry.get("passwd", []))
+
+            yield FortiOSUserRecord(
+                name=username,
+                password=password,
+                groups=",".join(local_groups.get(username, [])),
+                home=None,
+                _target=self.target,
+            )
+
+        # Temporary guest users
+        for _, entry in self._config["root-config"]["user"]["group"].get("guestgroup", {}).get("guest", {}).items():
+            try:
+                password = decrypt_password(entry.get("password")[-1])
+            except ValueError:
+                password = ":".join(entry.get("password"))
+
+            yield FortiOSUserRecord(
+                name=entry["user-id"][0],
+                password=password,
+                groups="guestgroup",
+                home=None,
+                _target=self.target,
+            )
 
-    @export(record=UnixUserRecord)
-    def users(self) -> Iterator[UnixUserRecord]:
-        # TODO: Add FortiOS specific users
-        return super().users()
+    @export(property=True)
+    def dns(self) -> list[str]:
+        """Return configured WAN DNS servers"""
+        entries = []
+        for _, entry in self._config["global-config"]["system"]["dns"].items():
+            entries.append(entry[0])
+        return entries
+
+    @export(property=True)
+    def install_date(self) -> Optional[str]:
+        """Return the likely install date of the operating system"""
+        if (init_log := self.target.fs.path("/data/etc/cloudinit.log")).exists():
+            return ts.from_unix(init_log.stat().st_mtime)
+
+    @export(property=True)
+    def timezone(self):
```

Comment:
Same with this and `language`, can you move this to a `locale.py`? And also add return types then.

Reply:
Added in [e765ee2](https://github.com/fox-it/dissect.target/pull/509/commits/e765ee29af0ff2bb8f524c6c5eb01eb7932f9ff6). See below for discussion.